src/load_gdm.cpp

```diff
@@ -0,0 +1,461 @@
+/*
+ * This source code is public domain.
+ *
+ * Authors: Alice Rowan <petrifiedrowan@gmail.com>
+*/
+
+////////////////////////////////////////////////////////////
+// General Digital Music module loader
+////////////////////////////////////////////////////////////
+
+#include "stdafx.h"
+#include "sndfile.h"
+
+static const DWORD GDM_SIG = 0xfe4d4447; // GDM\xFE
+static const DWORD GMFS_SIG = 0x53464d47; // GMFS
+
+typedef struct tagFILEHEADERGDM
+{
+	BYTE sig[4];		// GDM\xFE
+	char name[32];
+	char author[32];
+	BYTE eof[3];		// \x0D\x0A\x1A
+	BYTE sig2[4];		// GMFS
+	BYTE gdm_ver_major;
+	BYTE gdm_ver_minor;
+	BYTE tracker_id[2];
+	BYTE tracker_ver_major;
+	BYTE tracker_ver_minor;
+	BYTE panning[32];
+	BYTE globalvol;
+	BYTE default_speed;
+	BYTE default_bpm;
+	BYTE origfmt[2];
+	BYTE ordersPos[4];
+	BYTE nOrders;
+	BYTE patternsPos[4];
+	BYTE nPatterns;
+	BYTE samplesPos[4];
+	BYTE sampleDataPos[4];
+	BYTE nSamples;
+	BYTE messagePos[4];
+	BYTE messageLength[4];
+	BYTE ignore[12];
+} FILEHEADERGDM;
+
+typedef struct tagSAMPLEGDM
+{
+	enum SAMPLEGDMFLAGS
+	{
+		S_LOOP = (1<<0),
+		S_16BIT = (1<<1),
+		S_VOLUME = (1<<2),
+		S_PAN = (1<<3),
+		S_LZW = (1<<4),
+		S_STEREO = (1<<5)
+	};
+
+	char name[32];
+	char dosname[12];
+	BYTE ignore;
+	BYTE length[4];
+	BYTE loopstart[4];
+	BYTE loopend[4];
+	BYTE flags;
+	BYTE c4rate[2];
+	BYTE volume;
+	BYTE panning;
+} SAMPLEGDM;
+
+static WORD fixu16(const BYTE *val)
+{
+	return (val[1] << 8) | val[0];
+}
+
+static DWORD fixu32(const BYTE (&val)[4])
```

Comment:
Compile-tested in my fork, Watcom had trouble with this one:
```
load_gdm.cpp(235): Error! E473: col(20) function argument(s) do not match those in prototype
load_gdm.cpp(235): Note! N392: col(20) definition: 'unsigned fixu32( unsigned char const (& )[4])'
load_gdm.cpp(235): Note! N643: col(20) cannot convert argument 1 defined in: load_gdm.cpp(235) (col 21)
load_gdm.cpp(235): Note! N630: col(20) source conversion type is 'unsigned char (const )[4] (lvalue)'
load_gdm.cpp(235): Note! N631: col(20) target conversion type is 'unsigned char const [4] (lvalue)'
[many other similar errors]
```
Changing it to `static DWORD fixu32(const BYTE val[4])` fixed build without warnings.

Reply:
Weird, Watcom doesn't support array references? Changing this isn't a big deal, will do.

Reply:
OK, I replaced the array reference usage in `fixu32` with the recommended `const BYTE val[4]`.

Reply:
Thanks.